src/components/NavItem.tsx

```diff
@@ -1,32 +1,31 @@
 import { ComponentType } from "react";
 import Link from "next/link";
 
+import { tw } from "~/merge-classNames";
+
 interface NavItemProps {
   href: string;
   name: string;
   navIcon: ComponentType<{ className?: string }>;
   onClick: () => void;
   selected: boolean;
 }
-function classNames(...classes: string[]) {
-  return classes.filter(Boolean).join(" ");
-}
 
-export default function NavItem({ href, name, navIcon, onClick, selected }: NavItemProps) {
+export function NavItem({ href, name, navIcon, onClick, selected }: NavItemProps) {
   const Icon = navIcon;
   return (
     <Link
       href={href}
-      className={classNames(
-        selected ? "bg-gray-900 text-white" : "hover:bg-gray-900 hover:text-white",
-        "group flex gap-x-3 rounded-md p-2 text-sm font-semibold leading-6",
+      className={tw(
+        selected ? "bg-gray-900 text-white" : "",
+        "group flex gap-x-3 rounded-md p-2 text-sm font-semibold leading-6 hover:bg-gray-900 hover:text-white",
```

Comment:
the way we approach tw is that we define the first argument as the default stylings and then (in this case) if the selected prop is truthy we overwrite or add styles. In other words please flip these two arguments around :)